src/tools/perf/libperf.c

```diff
@@ -170,10 +247,71 @@ void ucx_perf_test_start_clock(ucx_perf_context_t *perf)
     perf->prev.time         = perf->start_time;
 }
 
+static ucs_status_t setup_cuda(ucx_perf_context_t *perf, int group_index)
+{
+    int         dev_count;
+    CUdevice    device;
+    CUresult    err;
+
+    if (perf->ctx_set) return UCS_OK;
+
+    err = cuInit(0);
+    if (CUDA_SUCCESS != err) {
+        fprintf(stderr, "error in cuInit\n");
+        exit(-1);
+    }
+    err = cuDeviceGetCount(&dev_count);
+    if (CUDA_SUCCESS != err) {
+        fprintf(stderr, "error in cuDeviceGetCount");
+        goto err;
+    }
+
+    if (dev_count < 2) {
```

Comment:
I think we should not  make test specific intra node

Reply:
I agree. I had this condition in place to check cudaipc functionality.